elk/utils.py

```diff
@@ -1,4 +1,33 @@
+from torch import Tensor
+from torch.nn.parallel import DistributedDataParallel as DDP
 from typing import Callable, Mapping, TypeVar
+import torch.distributed as dist
+import torch.nn as nn
+
+
+def maybe_all_gather(x: Tensor) -> Tensor:
```

Comment:
probably a dumb quesiton, but why the "maybe"s? Is it because you have maybe multiple processes or maybe just one?

Reply:
Yeah it's because these methods do nothing when the script isn't run with `torchrun`